app/assets/javascripts/index.js

```diff
@@ -61,25 +90,8 @@ $(document).ready(function () {
       dataType: "html",
       complete: function (xhr) {
         clearTimeout(loaderTimeout);
-        $("#flash").empty();
         $("#sidebar_loader").hide();
-
-        var content = $(xhr.responseText);
-
-        if (xhr.getResponseHeader("X-Page-Title")) {
-          var title = xhr.getResponseHeader("X-Page-Title");
-          document.title = decodeURIComponent(title);
-        }
-
-        $("head")
-          .find("link[type=\"application/atom+xml\"]")
-          .remove();
-
-        $("head")
-          .append(content.filter("link[type=\"application/atom+xml\"]"));
-
-        $("#sidebar_content").html(content.not("link[type=\"application/atom+xml\"]"));
-
+        OSM.setSidebarContent([xhr.getResponseHeader("X-Page-Title"), xhr.responseText]);
```

Comment:
What's the idea between packing these arguments into an array and then unpacking them inside the function rather than just passing two arguments?

Is it so you can pass the array returned by the get function into the set function in a later patch?

Reply:
The idea is for the get function to be able to receive as an argument whatever was returned by the set function.

Get() needs to return two things. They need to be packed into something before being returned. Currently it's easy to imagine how you can unpack them and pass as individual arguments to set(). What if I add more things? What if I decide that with more things it's better to return an object from get()? I don't want to unpack such objects for set() arguments.